File: Medium Problem's/p-204. Count Primes/Solution.js
/**
 * @param {number} n
 * @return {number}
 */
//Satyam-R4j-LeetCode Problem-204 (medium)
//Solution in JavaScript
var countPrimes = function (n)
{
    let isPrime = new Array(n + 1)
    isPrime.fill(true)
    let count = 0

    for (let i = 2; i < n; i++)
    {
        if (isPrime[i])
        {
            count++
            for (let j = i * 2; j < n; j = j + i)
            {
                isPrime[j] = false
            }
        }
    }

    return count

};

let n = 13
console.log(countPrimes(n))

File: Medium Problem's/p-204. Count Primes/Solution.py
import math


class Solution:
    def fillSieve(self, sieve):
        n = len(sieve)
        for i in range(2, int(math.sqrt(n)) + 1):
            if sieve[i] == 1:
                for j in range(i * i, n, i):
                    sieve[j] = 0

    def countPrimes(self, n: int) -> int:  # type: ignore
        count = 0
        sieve = [1] * n
        self.fillSieve(sieve)
        if n > 0:
            sieve[0] = 0
        if n > 1:
            sieve[1] = 0

        for i in range(2, n):
            if sieve[i] == 1:
                count += 1
        return count
